nova/scheduler/mixins.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2021 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import time
+
+from nova import context
+from nova.scheduler import client
+
+
+class HypervisorSizeFilterMixin(object):
+
+    _HV_SIZE_CACHE_RETENTION_TIME = 10 * 60
+
+    def __init__(self):
+        self._HV_SIZE_CACHE = {}
+
+    def _get_hv_size(self, host_state):
+        # expire the cache 10min after last write
+        time_diff = time.time() - self._HV_SIZE_CACHE.get('last_modified', 0)
```

Comment:
What is the motivation for this?
Personally,y I prefer an invalidation on an item level. Just because I've updated another host, doesn't mean that the current host will be valid, no?

Reply:
Valid point. This is just copied over and i can fix it as a follow-up. It was built out of simplicity to invalidate the cache at some point in time at least.

We have the same algorithm - which is even more stupid there - for the `ShardFilter`. This needs fixing, too.